src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,201 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a list of tasks with various operations such as marking, unmarking, listing, removing and adding tasks.
+ * Provides methods to interact with and manipulate the list of tasks.
+ */
+
+public class TaskList {
+
+    private ArrayList<Task> myList;
+
+    /**
+     * Constructs a TaskList instance with the provided list of tasks.
+     *
+     * @param myList The list of tasks to be managed.
+     */
+    public TaskList(ArrayList<Task> myList) {
+        this.myList = myList;
+    }
+
+    /**
+     * Marks a specified task in the list as done.
+     *
+     * @param parts An array containing the command and task number.
+     */
+    public void markList(String[] parts) {
+        //System.out.println("");
+        int index = Integer.parseInt(parts[1]) - 1;
+        Ui ui = new Ui();
```

Comment:
Similar to the TaskList instances, instead of creating a new Ui instance within each method, consider creating a single instance in the constructor and using it throughout the class.

Reply:
done!